### ISSUE TITLE
Cannot use GLUT_CORE_PROFILE on macOS 10.13.1

### ISSUE THREAD

**Issue by powerfulTrouser** [FreeGLUTProject/freeglut#63]
I was new for OpenGL, but when I tried to use GLUT_CORE_PROFILE in function glutInitDisplayMode,error messeges came out:

<img width="1280" alt="default" src="https://user-images.githubusercontent.com/20932259/33526144-881b93ee-d877-11e7-96d1-00f546a3982e.png">

Why this came out and how to solve this problem?

**Comment by rcmaniac25** [comment]
GLUT_CORE_PROFILE should be set to in glutInitContextProfile, not glutInitDisplayMode. In glutInitDisplayMode, GLUT_CORE_PROFILE is the same as GLUT_INDEX... and virtually no platforms support that value.

**Comment by nothodo** [comment]
<img width="1280" alt="default" src="https://user-images.githubusercontent.com/20932259/33526477-3dba42b8-d87d-11e7-826f-9f800e431627.png">

I think I mixed it up with glut....
But I still cannot use core OpenGL by adding that line
I'm using macbook pro 2015 and High Sierra, so it should support OpenGL 4.1.
Is freeglut not supporting core OpenGL or not supporting core OpenGL just on mac?
Or there are something I haven't done yet?

By the way, it can display correctly by using glut, but it's not using xquartz to display.
Is that matters?

<img width="1216" alt="default" src="https://user-images.githubusercontent.com/20932259/33526667-7e0534e8-d87f-11e7-8477-d977cba10a97.png">

**Comment by nothodo** [comment]
It does not work too.

<img width="1280" alt="default" src="https://user-images.githubusercontent.com/20932259/33551557-ae43e9b2-d92c-11e7-806a-6bc6fd96027d.png">

uncomment the line "glutInitContextProfile(GLUT_CORE_PROFILE);" has the same result.

**Comment by rcmaniac25** [comment]
I'm confused on what you tried: You tried it with the context version AND context profile at the same time? Or just one or the other?

**Comment by nothodo** [comment]
I had tried all possibility.

**Comment by rcmaniac25** [comment]
Ok. So the challenge for me: freeglut supports X11, including core profile as far as I know. On Mac, freeglut uses X11. But if Mac doesn't support Core on X11, then it won't work.

One thing to try, add a warning handler and see if anything shows up. If nothing shows up, then I'm stumped. I don't have a Mac to try things out on.

**Comment by Superlokkus** [comment]
I'm using freeglut @3.0.0 via macports and cmake on macOS 10.13 and think I got the same problem:
It can't link `_glutInitContextProfile` and `_glutInitContextVersion` but the other freeglut symbols link fine. When comparing to the macOS delivered header it seems obvious: These were removed as it seems.

**Comment by rcmaniac25** [comment]
That's different from what OP asked, where the requested feature he wants to use isn't functioning, as in a run-time issue

Your issue is a compile-time issue, but none of the symbols should start with underscores, which could be why you're having problems. So it should be `glutInitContextProfile` not `_glutInitContextProfile`

**Comment by jtsiomb** [comment]
I just tested creating a core profile context with freeglut on macos and it works fine. Either this was an issue with an older version of macos, or the OP requested an opengl version unsupported by macos (I tried with 3.3 core, not 4.1). Either way it's not a freeglut bug.

**Comment by jtsiomb** [comment]
Calling `glutInitContextFlags` twice has the effect of the second call overwriting whatever you've set on the first call. You need to call it once, with all the bitflags you care to set or-ed together.

**Comment by avegetablechicken** [comment]
> 

sorry it should be
```
glutInitContextProfile(GLUT_CORE_PROFILE);
glutInitContextFlags(GLUT_FORWARD_COMPATIBLE);
```
I tried either of the two line and both of them but still failed.

**Comment by avegetablechicken** [comment]
> Calling `glutInitContextFlags` twice has the effect of the second call overwriting whatever you've set on the first call. You need to call it once, with all the bitflags you care to set or-ed together.

sorry it should be

```
glutInitContextProfile(GLUT_CORE_PROFILE);
glutInitContextFlags(GLUT_FORWARD_COMPATIBLE);
```
I tried either of the two line and both of them but still failed.

**Comment by jtsiomb** [comment]
I booted the only computer I have with macos and tested again, and indeed it appears I'm not getting a core profile. The context is created but the version strings show 2.1 and `glBegin` still works. I must assume that previously I only tested that I got a context, and assumed it was the one I asked for, without querying the version.

Having said that, I'm still convinced this is not a bug in freeglut. Versioned context creation works perfectly fine on other UNIX systems, so I can only assume that the macos X11/GLX implementation does not support it.
That it works with glfw is not an indication of anything. glfw (I assume) has native macos support without relying on X11.
